### ISSUE TITLE
Trigger Alerts 2 days after last date of usage period

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#656]
In GitLab by @kdjstudios on Jan 25, 2018, 16:19

**Submitted by:** "Cori Bartlett" <cori.bartlett@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2018-01-25-75141/conversation

**HD:** http://www.servicedesk.answernet.com/profiles/ticket/2019-09-10-48638

**Server:** Internal

**Client/Site:** NA

**Account:** NA

**Issue:** 

We need an email alert sent out 2 days after the billing cycle closes if the new billing cycle is not finalized. The day the billing cycle closes is the last day of the usage period.

 

Please email a message like: “Notice that Billing cycle is not finalized for _____ invoice for _______ site.” Email notification should go to Accountingmanagers@answernet.com & cori.bartlett@answernet.com for every site. In addition, the site specific emails are as follows:

 

 

Allentown:          Allentownmanangers@answernet.com

Chicago:               Chicagomanagers@answernet.com

Atlanta:                Atlantamanagers@answernet.com

Birmingham, Chattanooga, Nashville: Triadmanagers@answernet.com

Billerica (IB) & (OB): Billericamanagers@answernet.com

Billings: Billingsmanagers@answernet.com

Bismarck: Bismarckmanagers@answernet.com

Columbus: Columbusmanagers@answernet.com

El Paso: Elpasomanagers@answernet.com

Energy Choice California: Linda.herner@answernet.com; jessica.brooks@answernet.com

Fair Oaks: fairoaksmanagers@answernet.com

Fairlawn: fairlawnmanagers@answernet.com

Hays: Haysmanagers@answernet.com

Lowell: Lowellmanangers@answernet.com

Memphis: Memphismanagers@answernet.com

Orlando: Orlandomanagers@answernet.com

Phoenix: phoenixmanagers@answernet.com

Portland: portlandmanagers@answernet.com

Rockland: Rocklandmanagers@answernet.com

Santa Rosa: Santarosamanagers@answernet.com

Sarasota: Sarasotamanagers@answernet.com

Stockton: stocktonmanagers@answernet.com

Toronto: Torontomanagers@answernet.com

Webster City: Webstercitymanagers@answernet.com

West Memphis: westmemphismanagers@answernet.com

Willow Grove: willowgrovemanagers@answernet.com

Winnipeg: Winnipegmanagers@answernet.com

**For Only Development Team** -  
* Add the above emails on Site alert email field
* Mark active False for the sites - 019, 028, 042, 120, 012, DEMO, 119, 073B, 034, TGEN, 035, 073, 106

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 31, 2018, 01:04

@kdjstudios 

we have this functionality, which sends out mail to the accounting team whenever a site does it's billing cycle process but, this one is different.

marking for discussion with @tim.traylor

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Mar 7, 2018, 03:47

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 30, 2018, 14:43

Updating ticket.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 30, 2018, 12:03

@smasih @abakshi @pchaudhary @tim.traylor 

Please bring this up in your morning call with @tim.traylor to discuss and update the priority, it has been over 4 months.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 26, 2019, 09:04

Marking this for an estimate, @tim.traylor believes this could be an cronjob,

I will want to follow up with Cori that this is still something they are wanting. If so, check if this needs to be a one time notification, or a continual (hourly/daily/etc) notification until it is solved.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 6, 2019, 15:31

I forgot to click send on a draft, so I just now wrote:

> We apologize for the delay. The support team is now being able to reference this ticket. We are just confirming if this request is still valid. 
> 
> If so, we would like to clarify, if this would just be a one time notification; or a continual notification until the billing is finalized. If continual, how often (Hourly, daily, weekly, etc)?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 18, 2019, 07:13

@kdjstudios 

Thank you for your approval.


We have noticed that there are no sites of the External Server as you mentioned above sites. 

Should we need to develop this functionality for external server too? If yes then provide the email address for each site.

Also, we noticed that All of the sites are not present which you provided a list of the site above.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 10, 2019, 09:13

Cori wrote:

> Hi Kyle- Yes, this is still valid. I think a notification 2 days after the cycle is supposed to close and every day thereafter until closed. We'd like a notification sent once the cycle is finalized as well. Rich?

Rich wrote:

> Agree:

I wrote:

> We will update the support team and keep you informed on the progress. 

@smasih @amishra @pchaudhary @tim.traylor 

Marked for estimate. Based upon Cori's feedback this would need to be a daily recurring notification.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 13, 2019, 04:57

@kdjstudios 


Thank you for the update.

Below is the estimate to send the notification.

Estimate:

|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|2|Worked with Concepts, implementation ok|3|Low Impact|1.0|4|

**Bifurcation:**

* Create a new attribute called(**Email Notification**) at the site level for storing all of the above-mentioned email address. (Only For Admin)
* Create a Script
* Setup on the cron job.
* Testing.

If you have any other suggestion on my first point please let us know.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 13, 2019, 12:27

@amishra 

Thank you for the estimate. This seems reasonable. Approving this for development.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 18, 2019, 04:55

changed time estimate to 4h

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 18, 2019, 07:20

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 18, 2019, 07:21

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 18, 2019, 08:37

@tim.traylor 

@amishra presented a great question, I would believe we will want to develop this for external too and allow them the choice to input a email or not to input an email.

Also I see in the estimate above it is marked only admins would have access to this at the site level. Should we actually be putting this in the email section in the actual admin section? Or what level of permission should have access to this at the site level, especially if we are providing this to our externals?

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 18, 2019, 09:06

@kdjstudios Is this another separate email address being added in the UI or is it just a checkbox to perform that action or not?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 18, 2019, 09:54

@tim.traylor 

Originally requested it is another separate email address, but we may also want to as you mention to include a checkbox in conjunction with that. Or maybe if there is no email address provided to not send then?

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 18, 2019, 12:27

@kdjstudios If they needed a new email address then yes, having them leave it blank means they don't want the notification.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 20, 2019, 07:22

@kdjstudios 

* We have created a cron for the required functionality but will let you know some points. Currently, there are many billing cycles which comes under this scenario.

> Please email a message like: “Notice that Billing cycle is not finalized for _____ invoice for _______ site.” Email notification should go to Accountingmanagers@answernet.com & cori.bartlett@answernet.com for every site.

* Subject: Billing Cycle Finalization Alert
* Body: Notice that Billing cycle is not finalized for (*No. of Invoices*) invoice for (*site name*) site.

* We will directly push this into the production servers in the upcoming Milestone.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 20, 2019, 08:54

@pchaudhary @tim.traylor 

Why am I not being allowed to test this on development?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 20, 2019, 09:30

@kdjstudios 

Could you please clarify to me what do you want to test so that I can help you for that because if we allow sending the mail on the development server, then the email for the same thing will send to all of the respected email ID's.
 **OR**

Would you like to add only your email ID for the development servers?

* We also would like to ask the timing of cron you want us to update for this on the development server?


* Although, you can check the Email field added on the site level for this functionality (Merge2) as suggested.

![image](/uploads/afc15d78175a97242267ba01b30e5773/image.png)

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 20, 2019, 09:30

assigned to @pchaudhary and unassigned @amishra

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 20, 2019, 11:00

@pchaudhary @tim.traylor 

I would like to be able to test this to confirm it works? I find it hard to believe this was tested, as two days has not passed.

I would believe I will need to create and draft a billing cycle, wait two days and confirm the email was sent? Then also wait a few more days to confirm the email is set again each day after. Then finalize the billing cycle and confirm that the next day no email is sent. How else would this be tested?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 20, 2019, 23:00

@kdjstudios 

We had tried to adjust the timing of cron according to our day & the emails were sent accordingly, so we supposed that it will trigger on the daily basis as we just have to change the time of cron.

Also, for the open billing cycle, we have got mail for each open billing cycle till now. That's what we have mentioned that there are multiple open billing cycles at the moment.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 21, 2019, 08:14

@pchaudhary @amishra @smasih @tim.traylor 

Lets actually test this with the correct settings. Please set this up on a dev server that we can leave it on for a few days to test this.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 22, 2019, 04:22

@kdjstudios 

Okay, then please let us know the timings on which you want this to be run?  
Also, we want to ensure that while testing this the alert mail will be sent to the above-mentioned Ids of the respected sites.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 22, 2019, 13:25

@pchaudhary @tim.traylor 

We need it to run on the specified timings in the ticket.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 23, 2019, 23:18

@kdjstudios @tim.traylor 

> We need it to run on the specified timings in the ticket.

* We need to know the timings on which you want the email hits. The timings mentioned in the tickets are for the day after the usage period ends.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 24, 2019, 08:58

@pchaudhary @tim.traylor 

Please have this run at 8am ET.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 25, 2019, 01:09

@kdjstudios 

We have updated the timing of the cron to send the alert email at 08:00 AM on the Merge2 server & updated the `apperrors@sahosted.com` as the recipient for testing purpose.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 25, 2019, 10:22

@pchaudhary @amishra @smasih @tim.traylor 

Which server did you push this too? Please be sure we are testing functionality after pushing things to a server, this continues to happen with the wrong branches being released.

I do not see any email address to enter on the site edit page? http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/656#note_42635

![image](/uploads/cf4a7058398d7c315aa50c50b3c80fda/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 26, 2019, 04:43

@kdjstudios 

Sorry for the inconvenience caused to you.   
We have updated the timing of the cron to send the alert email at 08:00 AM on the Merge2 server & updated the `apperrors@sahosted.com` as the recipient for testing purpose.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 26, 2019, 10:33

@pchaudhary 

Thank you, I have created and applied the billing cycle for the Demo site on Merge2. I have set my email address on the site edit page. and now it is at the finalize step. 

I just want to confirm that I should expect to receive the email alert on the 28th at 8AM ET, which is two days after today the 26th.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 26, 2019, 23:57

@kdjstudios 

We would like to mention here that we have not passed the Site alert email ID to the recipient of this email as if we will allow doing that then all of the emails will go to the real Email-Ids of the client before releasing it. So, we have mentioned the `apperrors@sahosted.com` for all of the sites from the background so that we can check the cron & mail subject & body.

* We request you to make it as an announcement as we can good to release this.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 27, 2019, 08:38

@pchaudhary @tim.traylor @smasih 

How was this even tested then? Why would we have real emails on a dev server? Sounds like you did not complete the development of these and then tried to mark it directly for ready a week ago http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/656#note_42615?

As stated in my comments above, we need to test this.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 27, 2019, 09:42

This was discussed today, @tim.traylor requested that @pchaudhary clarify which items were tested and how they were so I may verify them, This will be in place of me actually testing this ticket.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 1, 2019, 08:49

@amishra @smasih @pchaudhary @tim.traylor 

I have not seen a response on how to confirm this was tested, but based upon the amount of emails over the weekend, this is certainly seems to not be working. I received a few that show a billing cycle with no invoices? Also some of these are from 2017 that does not make sense either?

![image](/uploads/e62d5d26312b299b7bed7187a2effb07/image.png)


Does this mean we still have open billing cycles from 2017? in the mix or what is going on?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 2, 2019, 05:28

@kdjstudios 

We have set up the full testing branch for you now on the Merge2 server.   
* We have uploaded the latest DB of the internal production server to the Merge2 server so that the testing has the live data.
* There is no email address is filled on the Billing Cycle Alert field for any of the sites so that you can test by your own email ID but yeah we have added `apperrors@sahosted.com` as the default recipient for all sites.  
* 

> I received a few that show a billing cycle with no invoices

* Yes, there was nowhere specified for the number of invoices on the billing cycles, so the query is only hitting if the billing cycle end date has been passed 2 days ago.



> Also some of these are from 2017 that does not make sense either?

* As requested, 2 days after from the date of Billing Cycle usage period ends until the Billing Cycle finalized, the email hits daily, so it is.


> Does this mean we still have open billing cycles from 2017? in the mix or what is going on?

* Yes, you can easily see the open billing cycles by landing to the Billing Cycle Index Page.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 2, 2019, 11:10

@pchaudhary @tim.traylor 

As discussed today, we will need to exclude the non active sites from these emails. These will be the same as the ones being excluded from the reports with the 'all sites' functionality. But also will now include the Chicago site too, as it has been merged into the Dyad.

I can confirm the emails are being sent. So once the above is done please mark into the announcements.  

EDIT:

We also need to close old billing cycles too, as there appears to be billing cycles that have been generated after and the user/ops can no longer close themselves. May we please get a list of all those billing cycles. Example below shows both the US and Master billing cycles from 2015 being open, but we already have closed billing cycles after those. 

![image](/uploads/a046bf2eb880410b3698c9da39ce31a0/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 5, 2019, 01:31

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 5, 2019, 01:31

@kdjstudios 

We have successfully changed the alert to be run for only the active sites included Chicago as well.  

So marking it as announcement.

> We also need to close old billing cycles too, as there appear to be billing cycles that have been generated after and the user/ops can no longer close themselves. May we please get a list of all those billing cycles. The example below shows both the US and Master billing cycles from 2015 is open, but we already have closed billing cycles after those.

* Please find the list of open Billing Cycles for both of the servers.
  * Internal - 52 Billing Cycles [previous_non_finalized_billing_internal.csv](/uploads/a9bd328d641d6ada2ba22eac73a044e0/previous_non_finalized_billing_internal.csv)
  * External - 2 Billing Cycles [previous_non_finalized_billing_hosted.csv](/uploads/0784b2e4e4eab2279f153db01574a768/previous_non_finalized_billing_hosted.csv)

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 5, 2019, 01:32

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 5, 2019, 02:19

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 5, 2019, 09:57

@tim.traylor 

With your approval, we should be able to have the Aavaz team mark those old billing cycles and completed/finalized.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 5, 2019, 10:05

@pchaudhary 

I reviewed this with @tim.traylor and rather then finalizing old billing cycles; which may have negative effects, we just want to add a start date filter to this. So any billing cycle after 7/1/19 is eligible for these alerts; But any billing cycles before this are not. Marking back for development. Please post an updated estimate as needed.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 8, 2019, 08:21

@pchaudhary 

Please post an updated estimate if it is going to take more then 1 hour to complete the above requested.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 29, 2019, 04:56

@kdjstudios 

We have developed the functionality as requested.

* All of the below Site Numbers are marked as non-active from the backend.
  * 019
  * 028
  * 042
  * 120
  * 012
  * DEMO
  * 119
  * 073B
  * 034
  * TGEN
  * 035
  * 073 
  * 106

* The cron to send the Billing Cycle alert will run at **08:00 AM EDT** 

* Please feel free to add the billing cycle alert email on the Site edit page.

* Although, we have added `apperrors@sahosted.com` as the default recipient.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 5, 2019, 14:44

I was able to confirm receipt of the emails. Marking this for announcement.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Aug 31, 2019, 04:41

@kdjstudios 

We have successfully released this for both of the production servers.

* Please note, we have changed the apperrors email to the criticalerrors email for the above cron.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 31, 2019, 12:05

@pchaudhary 

Thank you for releasing the code, however you completely failed to assign the emails as specified in the request. 

I wrote:

> The support team was able to develop the alerts and have released this to SA Billing. There is now a "Billing cycle alert email" field on the site edit page where operations can specify the email address to receive these alerts at. Please let us know if there are any questions or if we may close this ticket. Thank you and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 6, 2019, 08:42

@amishra @pchaudhary @smasih @tim.traylor 

Marking back for development. It would appear this was setup for only admins to access as we never got a clear response on http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/656#note_42434. (NOTE: I don't know where this was ever requested to only be for admins either too? Who made this decision?) This should have anyone with site level access should be able to access it accordingly, I believe. I will confirm with Cori.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 6, 2019, 08:43

Cori wrote:

> Kyle- It must not be live in production. See below.
> 
> ![image](/uploads/7e0312088231f098f284ca1aed4da096/image.png)

I wrote:

> Thank you for notifying us that you are unable to see it. Rich are you seeing it?
> 
> Currently it is live in production as I am seeing it; but from your response Cori, I presume that the permission access to it is not correctly setup. Just to confirm what level of users should be able to see and edit this?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 6, 2019, 10:33

Cori wrote:

> whatever users are able to edit site settings. 

@amishra @pchaudhary 

Please update the permissions for this and let us know when it is ready for testing.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 10, 2019, 13:03

Cori wrote:

> Hi Kyle- When did this go live? Also, can we add more than 1 email address to the alert?


I wrote:

> This feature went live on the August 31st SA Billing release. Please note the permission issue you pointed out that following week with allowing non tech admins ability to edit the email address is still in development. We will keep those updates posted on the original ticket.
> 
> Let me check with the support team about multiple emails addresses; but I believe it was only developed with one email address per site. If that is the case, I presume you will probably want this to also be updated to allow for more then one email address, correct?

@amishra @pchaudhary @jredfield 

May we please have the permission fixed before this next release. Also may we please confirm if this allows for more then one email address? If not allowing for multiple, we will need to update that too.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 10, 2019, 13:17

@pchaudhary @tim.traylor @smasih @jredfield 

Who made the approval to change the email address on this? These should be going to apperrors@sahosted.com not to criticalerrors@sahosted.com. The critical error email should be used for when a cron job fails  or something with the server fails?

I see the following comment, but I do not see who requested and approved this.
http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/656#note_46628

Also Amanda sent this to me:

> Delivery has failed to these recipients:
> 
> billing_cycle@sabhosted.com
> 
> Subject: RE: Billing Cycle Finalization Alert

Where did that email come from? Looks to be the from address of the email we are sending out.

![image](/uploads/bcab91916bc9aa8a388fa2fc8f532416/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 10, 2019, 13:55

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 11, 2019, 01:53

@kdjstudios 

> Please update the permissions for this and let us know when it is ready for testing.

* We have changed the permission & it is ready for testing on **Merge2** server. 

> Also may we please confirm if this allows for more then one email address? If not allowing for multiple, we will need to update that too.

* We need to update the functionality accordingly.



> Who made the approval to change the email address on this? These should be going to apperrors@sahosted.com, not to criticalerrors@sahosted.com. The critical error email should be used for when a cron job fails or something with the server fails?

* Okay, we will change this too.

> Where did that email come from? Looks to be the from the address of the email we are sending out.

* This email came from the Billing Cycle Finalization alert.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 11, 2019, 09:00

@pchaudhary @jredfield 

Okay so we have quite a few updates that still need to occur before we send this back to Answernet for testing. Please estimate (if more then 2 hours are needed) and perform the following.

1. "From" address change to "support@sahosted.com"
2. "BCC" and "CC" address field remove all addresses currently in there.
3. "BCC" address field add only "apperrors@sahosted.com"
4. Billing cycle email field on site edit page, update to allow for more then one email address.
5. "To" field set to all the email addresses specified in the field notated in step 4 above.
6. Update permissions on the email field on the site edit page. 

Once all of the above is completed and tested please specify for Answernet to test.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 13, 2019, 05:41

@kdjstudios 

We have developed all of the above things on the development servers. We will update you once ready for your testing.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Sep 23, 2019, 23:16

@kdjstudios 

We have set up a branch on the merge2 server. Please test it.

### Email Screenshot

![email_screen](/uploads/ebea03e34f848921c0c435e6bf4d0073/email_screen.png)


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 24, 2019, 08:21

@amishra 

Thank you, I was able to confirm step 4 and 6. I will now open a billing cycle and wait for the emails to confirm they are sent accordingly from and to the correct addresses.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2019, 08:37

@amishra 

I did not receive any emails last night. This would appear to not be working. I should have received them for the Demo Site's NCSM billing cycle screenshot below. It should have been sent to both of my emails, along with the apperrors.

Billing Cycle:

![image](/uploads/a5af588f8a9f8ed8430267ca9758b32c/image.png)

Site:

![image](/uploads/abe4c4d2344312f3eb8e0c6bae7c2ad6/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2019, 08:37

removed assignee

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Sep 25, 2019, 22:30

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Sep 25, 2019, 23:07

@kdjstudios 

Thank you for the update.

We would like to inform you that we send finalization alert notification for only active site and demo site was marked for inactive.

So, for your testing purpose, we marked for active and now you are able to get an email notification. Once you confirm we will marked for inactive.

**Note: The cron to send the Billing Cycle alert will run at **08:00 AM EDT****

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 26, 2019, 08:36

@amishra 

Thank you! Works great. Marking for Ready

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Sep 28, 2019, 03:10

@kdjstudios 

We have successfully released on both of the production servers.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 28, 2019, 19:35

I wrote:

> The support team was able to release the updates to the permisions. Now anyone with edit to the sites should be able to adjust the Billing Cycle Email Alert. This also now allows for multiple email addresses. Please use a comma in between each address.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 1, 2019, 10:54

closed
